src/libstd/sys/unix/net.rs

```diff
@@ -378,21 +378,22 @@ impl IntoInner<c_int> for Socket {
 // against glibc version < 2.26. (That is, when we both know its needed and
 // believe it's thread-safe).
 #[cfg(target_env = "gnu")]
-pub fn res_init_if_glibc_before_2_26() -> io::Result<()> {
+pub fn on_resolver_failure(e: io::Error) -> io::Error {
     // If the version fails to parse, we treat it the same as "not glibc".
     if let Some(Ok(version_str)) = glibc_version_cstr().map(CStr::to_str) {
         if let Some(version) = parse_glibc_version(version_str) {
             if version < (2, 26) {
-                let ret = unsafe { libc::res_init() };
-                if ret != 0 {
-                    return Err(io::Error::last_os_error());
-                }
+                unsafe { libc::res_init() };
```

Comment:
Any particular reason why you chose to prefer the original resolver error over any potential `res_init` error?

Reply:
I considered this but:

1. That's what the code in `master` is doing. The error from `res_init_if_glibc_before_2_26` is discarded and the original error from `getaddrinfo()` is returned.

2.  Callers are probably expecting `getaddrinfo()`-type errors from `lookup_host()` (e.g. `EAI_FAIL`) . This is especially true since this will only be happening on an increasingly small fraction of platforms in obscure cases so it won't get much testing. I think returning the `res_init()` result here would be surprising.

3. `res_init()` does not return a well defined set of error codes. It's not clear that's it's setting `errno` at all from reading the glibc manage

4. I'm not convinced that `res_init()` will always be the most proximate error. It's possible that `res_init()` is returning a useful report on why the resolution system isn't working (but I have my doubts due to point 3). However, it's also possible that the `getaddrinfo()` is returning more useful information and the `res_init()` is failing for some inconsequential reason.

I kept the error code return from `on_resolver_failure` in case some platform wants to modify the error on return but it's just directly returned at the moment.

If there's a strong opinion in the other direction I can change the error behaviour but I want to reiterate that would be a functional change from `master`.